### PR TITLE
Add misbehave reason to Misbehave

### PR DESCRIPTION
**Pull request by dagurval**
To simplify debugging

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -4956,7 +4958,7 @@ bool ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     else if (pfrom->nVersion == 0)
     {
         // Must have a version message before anything else
-        Misbehaving(pfrom->GetId(), 1);
+        Misbehaving(pfrom->GetId(), 1, "version not sent");
```
Comment (dgenr8) [inline src/main.cpp]: "no version received" ?

File: src/main.cpp [hunk 2]
```diff
@@ -5507,7 +5509,7 @@ bool ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
         if (!dontWant.IsWithinSizeConstraints())
             // There is no excuse for sending a too-large filter
-            Misbehaving(pfrom->GetId(), 100);
+            Misbehaving(pfrom->GetId(), 100, "get_xthin: too large filter");
```
Comment (dgenr8) [inline src/main.cpp]: "get_xthin: filter too large"

File: src/main.cpp [hunk 3]
```diff
@@ -5728,7 +5730,7 @@ bool ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
         if (!filter.IsWithinSizeConstraints())
             // There is no excuse for sending a too-large filter
-            Misbehaving(pfrom->GetId(), 100);
+            Misbehaving(pfrom->GetId(), 100, "filterload: too large filter");
```
Comment (dgenr8) [inline src/main.cpp]: filterload: filter too large

**Comment by dagurval** [comment]
Updated with suggested changes by @dgenr8

**Pull request merged**